com.unity.ml-agents/Runtime/Analytics/InferenceAnalytics.cs

```diff
@@ -0,0 +1,97 @@
+using System.Collections.Generic;
+using Unity.Barracuda;
+using Unity.MLAgents.Actuators;
+using Unity.MLAgents.Inference;
+using Unity.MLAgents.Policies;
+using Unity.MLAgents.Sensors;
+using UnityEditor;
+using UnityEditor.Analytics;
+using UnityEngine.Analytics;
+
+namespace Unity.MLAgents.Analytics
+{
+    internal class InferenceAnalytics
+    {
+        static bool s_EventRegistered = false;
+        const int k_MaxEventsPerHour = 1000;
+        const int k_MaxNumberOfElements = 1000;
+        const string k_VendorKey = "unity.ml-agents";
+        const string k_EventName = "InferenceModelSet";
+
+        static bool EnableAnalytics()
+        {
+            if (s_EventRegistered)
```

Comment:
This is adapted from the example [here](https://github.com/Unity-Technologies/ScriptableRenderPipeline/blob/2bdf0eb2a88ae6085b18979cb427b01c9bf862c8/com.unity.shadergraph/Editor/ShaderGraphAnalytics.cs#L16-L41) (internal link)